R/mod_reweightView.R

```diff
@@ -273,8 +276,17 @@ reweightViewServer <- function(id, user, decision_list) {
       rlang::inform(paste(wt_chgd_metric, ": ", wt_chgd_wt))
       purrr::walk2(wt_chgd_metric, wt_chgd_wt, ~update_metric_weight(.x, .y))
       
+      curr_new_wts(
+        get_metric_weights() %>%
+          dplyr::mutate(new_weight = weight) %>%
+          dplyr::mutate(weight = ifelse(name == "covr_coverage", 0, weight)))
+      
+      user$metrics_reweighted <- user$metrics_reweighted + 1
+      
       # update for each package
       all_pkgs <- dbSelect("SELECT DISTINCT name AS pkg_name FROM package")
+      req(nrow(all_pkgs) > 0)
```

Comment:
I would at least add a comment here saying why this is important and that deleting it will re-introduce a bug.

A unit test would make sense as well, but I can't recall where the dev team landed on the testing strategy (e.g. did we adopt the mantra that we add tests for 1) new functionality and 2) any bugs that are discovered and fixed to prevent regression?).

Reply:
Both good points. I will add both.

Reply:
> did we adopt the mantra that we add tests for 1) new functionality and 2) any bugs that are discovered and fixed to prevent regression?).

I think we landed on #1 for sure. But I like #2 also. Let's do both. I'll add it to our PR checklist wiki.